lib/pages/home_page.dart

```diff
@@ -54,7 +54,7 @@ class _HomePageState extends State<HomePage> {
                       settings: const RouteSettings(name: "SettingsPage"),
                       builder: (c) => const SettingsPage(),
                     ),
-                  );
+                  ).then((_) => setState(() {}));
```

Comment:
We can do this in a more idiomatic way by using an async function for onPressed (I think a little easier to follow):

```dart
        onPressed: () async {
                  await Navigator.push(
                    context,
                    MaterialPageRoute(
                      settings: const RouteSettings(name: "SettingsPage"),
                      builder: (c) => const SettingsPage(),
                    ),
                  );
                  // Refresh UI.
                  setState(() {});
            }
     )
```

Why do we need to refresh the UI though?

Reply:
This one is not working tho. My intention was to refresh UI that way when user goes to settings and log-in or log-out then back to home screen should make UI changes according to login status. Once we add provider, we don't even need this. I can remove it.